Guides/Codable-Guide.md

```diff
@@ -1,9 +1,21 @@
 # `Codable` Usage in MongoSwift and MongoSwiftSync
 There are a number of ways for users to leverage `Codable` via the driver's API. One such example is through `MongoCollection<T>`. By default, `MongoDatabase.collection` returns a `MongoCollection<BSONDocument>`. Any `find` or `aggregate` method invocation on that returned collection would then return a `MongoCursor<BSONDocument>`, which when iterated returns a `BSONDocument?`:
+
```

Comment:
since this guide and the README had short examples and already showed multiple versions (both futures and sync) I kept the future based examples and just added async/await versions too.